mne/cov.py

```diff
@@ -1366,8 +1336,17 @@ def prepare_noise_cov(noise_cov, info, ch_names=None, rank=None,
         and parameters updated.
     """
     # reorder C and info to match ch_names order
+    noise_cov_idx = list()
+    missing = list()
     ch_names = info['ch_names'] if ch_names is None else ch_names
-    noise_cov_idx = [noise_cov.ch_names.index(c) for c in ch_names]
+    for c in ch_names:
+        try:
```

Comment:
can you avoid the try except with

if c in noise_cov.ch_names:
   noise_cov_idx.append(noise_cov.ch_names.index(c))
else:
    missing.append(c)

Reply:
Yes but it actually goes against what Python recommends, and in this case is the same number of lines

Reply:
can I object to the Python recommendation then? it's more readable although it requires 2 pass on the list